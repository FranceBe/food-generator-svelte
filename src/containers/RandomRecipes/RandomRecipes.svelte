<script>
  import { writable } from 'svelte/store'
  import H2 from 'components/H2/H2.svelte'
  import ButtonBurgerShape from 'components/ButtonBurgerShape/ButtonBurgerShape.svelte'
  import RecipeCard from 'components/RecipeCard/RecipeCard.svelte'
  import { getRandomInt, getRandomNItems } from 'services/random.service'
  import ReceipeCard from '../../components/RecipeCard/RecipeCard.svelte'

  let recipes = []

  const listOfTotalRecipes = writable([])
  export const getRandomRecipes = n => {
    const randomPage = getRandomInt(50)
    fetch(`api/puppy/?p=${randomPage}`)
      .then(res => res.json())
      .then(data => {
          const listOfRecipes = getRandomNItems(data.results, n)
          listOfTotalRecipes.set(listOfRecipes)
      })
            .catch(err => {
                throw Error(err)
      })
  }
  listOfTotalRecipes.subscribe(value => {
    recipes = value
  })
</script>

<style>
  div {
      display: flex;
      flex-direction: row;
  }
  .global-container {
      display: flex;
      flex-direction: column;
      flex-wrap: wrap;
      justify-content: center;
      align-items: center;
      margin: 20px 0;
  }
</style>

<H2 title="Combien voulez vous de propositions ?"/>
<div>
  <ButtonBurgerShape className="with-marge" onClick={() => getRandomRecipes(1)} text="1"/>
  <ButtonBurgerShape className="with-marge" onClick={() => getRandomRecipes(5)} text="5"/>
  <ButtonBurgerShape onClick={() => getRandomRecipes(10)} text="10"/>
</div>
{#if recipes.length > 0}
<div class="global-container">
  {#each recipes as recipe}
    <ReceipeCard title={recipe.title} href={recipe.href} ingredients={recipe.ingredients} src={recipe.thumbnail}/>
  {/each}
</div>
{/if}
