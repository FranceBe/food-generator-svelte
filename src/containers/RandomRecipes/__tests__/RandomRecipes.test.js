import { render, fireEvent, cleanup } from '@testing-library/svelte'
import RandomRecipes from 'containers/RandomRecipes/RandomRecipes.svelte'

describe('RandomRecipes', () => {
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const { container } = render(RandomRecipes)
    expect(container).toMatchSnapshot()
  })
  it('should render h2', () => {
    const { container, getAllByText } = render(RandomRecipes)
    expect(container.getElementsByTagName('h2')).toHaveLength(1)
    expect(getAllByText('Combien voulez vous de propositions ?').length).toBe(1)
  })
  it('should render 2 div and 3 buttons', () => {
    const { container, getAllByText } = render(RandomRecipes)
    expect(container.getElementsByTagName('div')).toHaveLength(2)
    expect(container.getElementsByTagName('button')).toHaveLength(3)
    expect(getAllByText('1').length).toBe(1)
    expect(getAllByText('5').length).toBe(1)
    expect(getAllByText('10').length).toBe(1)
  })
  it('should call fetch when a button is clicked', async() => {
    const { getByText } = render(RandomRecipes)
    const button = getByText('1')

    await fireEvent.click(button)
    expect(fetch).toHaveBeenCalledTimes(1)
  })
})
