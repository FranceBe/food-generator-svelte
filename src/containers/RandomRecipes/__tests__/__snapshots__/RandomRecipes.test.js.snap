// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomRecipes should match snapshot 1`] = `
<body>
  <div>
    <h2>
      Combien voulez vous de propositions ?
    </h2>
     
    <div>
      <button
        class="with-marge"
      >
        <span>
          1
        </span>
      </button>
       
      <button
        class="with-marge"
      >
        <span>
          5
        </span>
      </button>
       
      <button
        class=""
      >
        <span>
          10
        </span>
      </button>
    </div>
     
    
  </div>
</body>
`;
